uniffi_core/src/ffi/foreignexecutor.rs

```diff
@@ -39,19 +39,40 @@ unsafe impl Sync for ForeignExecutorHandle {}
 /// bindings should release the reference to the executor that was reserved for Rust.
 ///
 /// This callback can be invoked from any thread, including threads created by Rust.
+///
+/// The callback should return one of the `FOREIGN_EXECUTOR_CALLBACK_*` codes;
 pub type ForeignExecutorCallback = extern "C" fn(
     executor: ForeignExecutorHandle,
     delay: u32,
     task: Option<RustTaskCallback>,
     task_data: *const (),
-);
+) -> i8;
+
+/// Returned by `ForeignExecutorCallback` if the callback was scheduled successfully
+pub const FOREIGN_EXECUTOR_CALLBACK_SUCCESS: i8 = 0;
+/// Returned by `ForeignExecutorCallback` if the callback couldn't be scheduled because the
+/// foreign executor is canceled/closed.
+pub const FOREIGN_EXECUTOR_CALLBACK_CANCELED: i8 = 1;
+/// Returned by `ForeignExecutorCallback` for all other errors
+pub const FOREIGN_EXECUTOR_CALLBACK_ERROR: i8 = 2;
```

Comment:
```suggestion
) -> u8;

/// Returned by `ForeignExecutorCallback` if the callback was scheduled successfully
pub const FOREIGN_EXECUTOR_CALLBACK_SUCCESS: u8 = 0;
/// Returned by `ForeignExecutorCallback` if the callback couldn't be scheduled because the
/// foreign executor is canceled/closed.
pub const FOREIGN_EXECUTOR_CALLBACK_CANCELED: u8 = 1;
/// Returned by `ForeignExecutorCallback` for all other errors
pub const FOREIGN_EXECUTOR_CALLBACK_ERROR: u8 = 2;
```